Flexible Learning/report_unitHistory.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/*
+Gibbon, Flexible & Open School System
+Copyright (C) 2010, Ross Parker
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use Gibbon\Forms\Form;
+use Gibbon\Services\Format;
+use Gibbon\Domain\User\UserGateway;
+use Gibbon\Forms\DatabaseFormFactory;
+use Gibbon\Domain\Students\StudentGateway;
+use Gibbon\Module\FlexibleLearning\Tables\UnitHistory;
+
+$highestAction = getHighestGroupedAction($guid, '/modules/Flexible Learning/report_unitHistory.php', $connection2);
+
+if (isActionAccessible($guid, $connection2, '/modules/Flexible Learning/report_unitHistory.php') == false) {
+    // Access denied
+    $page->addError(__('You do not have access to this action.'));
+} elseif (empty($highestAction)) {
+    // Check the action with highest precedence
+    $page->addError(__('The highest grouped action cannot be determined.'));
+} else {
+    // Proceed!
+    $gibbonSchoolYearID = $gibbon->session->get('gibbonSchoolYearID');
+    $studentGateway = $container->get(StudentGateway::class);
+
+    if ($highestAction == 'Unit History_all') {
+        // Can view all students
+        $page->breadcrumbs->add(__m('Unit History'));
```

Comment:
Would this not be better above line 40, and then removed from line 49? Seems like a duplication of the same code.